src/ray/rpc/server_call.h

```diff
@@ -369,19 +415,21 @@ class ServerCallFactoryImpl : public ServerCallFactory {
         cq_(cq),
         io_service_(io_service),
         call_name_(std::move(call_name)),
+        cluster_id_(cluster_id),
```

Comment:
This line has a different semantics when you change from a pointer to a reference. This uses a copy constructor which is defined in id.h and creates a new object.